ipi/engine/atoms.py

```diff
@@ -259,11 +252,17 @@ def get_kstress(self):
       tensor -- not volume-scaled
       """
 
-      ks = np.zeros((3,3),float)
-      ks[0,0] = np.dot(self.px,self.px/self.m)
-      ks[1,1] = np.dot(self.py,self.py/self.m)
-      ks[2,2] = np.dot(self.pz,self.pz/self.m)
-      ks[0,1] = np.dot(self.px,self.py/self.m)
-      ks[0,2] = np.dot(self.px,self.pz/self.m)
-      ks[1,2] = np.dot(self.py,self.pz/self.m)
+      p = depstrip(self.p)
+      m = depstrip(self.m)
```

Comment:
No, when you call depstrip it checks for the need of updating. by depstripping you just assume that while you use (without changing!) p and m, they won't be modified elsewhere.